frb_example/pure_dart/rust/src/api.rs

```diff
@@ -725,3 +725,111 @@ impl ConcatenateWith {
         Ok(())
     }
 }
+
+pub enum CustomError {
+    Error0(String),
+    Error1(u32),
+}
+
+pub fn return_err_custom_error() -> Result<u32, CustomError> {
+    Err(CustomError::Error1(3))
+}
+
+pub fn return_ok_custom_error() -> Result<u32, CustomError> {
+    Ok(3)
+}
+
+pub fn return_error_variant(variant: u32) -> Result<u32, CustomError> {
+    match variant {
+        0 => Err(CustomError::Error0("variant0".to_string())),
+        1 => Err(CustomError::Error1(1)),
+        _ => panic!("unsupported variant"),
+    }
+}
+
+pub struct SomeStruct {
+    pub value: u32,
+}
+
+impl SomeStruct {
+    pub fn new(value: u32) -> SomeStruct {
+        SomeStruct { value }
+    }
+    pub fn static_return_err_custom_error() -> Result<u32, CustomError> {
+        Err(CustomError::Error1(3))
+    }
+
+    pub fn static_return_ok_custom_error() -> Result<u32, CustomError> {
+        Ok(3)
+    }
+
+    pub fn non_static_return_err_custom_error(&self) -> Result<u32, CustomError> {
+        Err(CustomError::Error1(self.value))
+    }
+
+    pub fn non_static_return_ok_custom_error(&self) -> Result<u32, CustomError> {
+        Ok(self.value)
+    }
+}
+
+#[allow(dead_code)]
```

Comment:
uber-nit: instead of allowing dead code, shall we really use it

Reply:
Do you mean adding tests that return the unused variants? They were there just to see if it would work generating more variants. Should I write tests that use them?

Reply:
If it generates but we never test it, I guess maybe there is not an *automated* approach to test that it really exists and works...